mlir/lib/Dialect/QC/Translation/TranslateQASM3ToQC.cpp

```
@@ -0,0 +1,1037 @@
+        idExpr->indices[0]->indexExpressions[0], debugInfo);
+    if (idx >= regBits.size() || !regBits[idx]) {
+      throw qasm3::CompilerError(
+          "Bit " + std::to_string(idx) + " of register '" + regName +
+              "' was not measured before use in condition.",
+          debugInfo);
+    }
+    return regBits[idx];
+  }
+
+  /// Convert qc::ComparisonKind to arith::CmpIPredicate.
+  static arith::CmpIPredicate convertComparisonKind(::qc::ComparisonKind kind) {
```

Cppcheck (2.20.0)

[style] The function 'convertComparisonKind' is never used.

(unusedFunction)